File: app/src/test/java/coolpharaoh/tee/speicher/tea/timer/core/settings/SortModeTest.java
package coolpharaoh.tee.speicher.tea.timer.core.settings;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.Test;

class SortModeTest {

    @Test
    void getIndexVariety() {
        final SortMode sortMode = SortMode.BY_VARIETY;
        assertThat(sortMode.getChoice()).isEqualTo(2);
    }

    @Test
    void sortModeFromIndexTwo() {
        final SortMode sortMode = SortMode.fromChoice(2);
        assertThat(sortMode).isEqualTo(SortMode.BY_VARIETY);
    }

    @Test
    void temperatureUnitFromTextNotDefined() {
        final SortMode sortMode = SortMode.fromChoice(5);
        assertThat(sortMode).isEqualTo(SortMode.LAST_USED);
    }
}

File: app/src/test/java/coolpharaoh/tee/speicher/tea/timer/core/settings/SortModeTest.kt
package coolpharaoh.tee.speicher.tea.timer.core.settings

import coolpharaoh.tee.speicher.tea.timer.core.settings.SortMode.Companion.fromChoice
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

internal class SortModeTest {
    @Test
    fun getIndexVariety() {
        val sortMode = SortMode.BY_VARIETY
        assertThat(sortMode.choice).isEqualTo(2)
    }

    @Test
    fun sortModeFromIndexTwo() {
        val sortMode = fromChoice(2)
        assertThat(sortMode).isEqualTo(SortMode.BY_VARIETY)
    }

    @Test
    fun temperatureUnitFromTextNotDefined() {
        val sortMode = fromChoice(5)
        assertThat(sortMode).isEqualTo(SortMode.LAST_USED)
    }
}